internal/v1/handler_blueprints.go

```diff
@@ -107,3 +109,63 @@ func (h *Handlers) ComposeBlueprint(ctx echo.Context, id openapi_types.UUID) err
 
 	return ctx.JSON(http.StatusCreated, composeResponses)
 }
+
+func (h *Handlers) GetBlueprints(ctx echo.Context, params GetBlueprintsParams) error {
+	spec, err := GetSwagger()
+	if err != nil {
+		return err
+	}
+	idHeader, err := getIdentityHeader(ctx)
+	if err != nil {
+		return err
+	}
+
+	limit := 100
+	if params.Limit != nil {
+		if *params.Limit > 0 {
+			limit = *params.Limit
+		}
+	}
+
+	offset := 0
+	if params.Offset != nil {
+		offset = *params.Offset
+	}
+
+	blueprints, count, err := h.server.db.GetBlueprints(idHeader.Identity.OrgID, idHeader.Identity.AccountNumber, limit, offset)
+	if err != nil {
+		return err
+	}
+
+	data := make([]BlueprintItem, 0, len(blueprints))
+	for _, blueprint := range blueprints {
+		data = append(data, BlueprintItem{
+			Id:          blueprint.Id,
+			Name:        blueprint.Name,
+			Description: blueprint.Description,
+			Version:     blueprint.Version,
+			CreatedAt:   blueprint.CreatedAt.Format(time.RFC3339),
+		})
+	}
+	lastOffset := count - 1
+	if lastOffset < 0 {
+		lastOffset = 0
+	}
+	return ctx.JSON(http.StatusOK, BlueprintsResponse{
+		Meta: struct {
+			Count int `json:"count"`
+		}{
+			count,
+		},
+		Links: struct {
+			First string `json:"first"`
+			Last  string `json:"last"`
+		}{
+			fmt.Sprintf("%v/v%v/composes?offset=0&limit=%v",
+				RoutePrefix(), spec.Info.Version, limit),
+			fmt.Sprintf("%v/v%v/composes?offset=%v&limit=%v",
+				RoutePrefix(), spec.Info.Version, lastOffset, limit),
+		},
+		Data: data,
+	})
```

Comment:
I don’t like this but since there is prior art on this, I am okay.

There should be a common function somewhere which constructs metadata which should be common across all endpoints so no anonymous struct should be used. Also, `GetSwagger()` parses the whole OpenAPI specification _every single request_ just to get some version, this should be done in an `init()` just once at the start of the program.

Reply:
I don't like this either, and I agree that a common function is needed here, but it sounds like a refactor not for this scope.